k2/csrc/array.h

```diff
@@ -52,70 +56,150 @@ struct Array2Size {
   // `size1` is the top-level size of the array, equal to the object's .size
   // element
   I size1;
-  // `size2` is the nunber of elements in the array, equal to
+  // `size2` is the number of elements in the array, equal to
   // o->indexes[o->size] - o->indexes[0] (if the Array2 object o is
   // initialized).
   I size2;
 };
 
-template <typename I, typename Ptr>
+template <typename Ptr, typename I = int32_t>
 struct Array2 {
```

Comment:
@csukuangfj I don't define any constructor or `Init` method in `Array2`, you can merge yours if you think it will be convenient in Python code.